src/Apps/Auction/Components/RegistrationForm.tsx

```diff
@@ -0,0 +1,296 @@
+      </Box>
+      <Box mt={4}>
+        <Serif weight="semibold" size="4t">
+          Billing Address
+        </Serif>
+        <Box mt={2}>
+          <Input
+            onBlur={handleBlur}
+            onChange={handleChange}
+            error={touched.street1 && errors.street1}
+            required
+            title="Address"
+            name="street1"
+          />
+        </Box>
+        <Flex mt={2}>
+          <Flex flexBasis="50%" flexGrow="2" mr={1}>
+            <Input
+              onBlur={handleBlur}
+              onChange={handleChange}
+              error={touched.city && errors.city}
+              required
+              title="City"
+              name="city"
+            />
+          </Flex>
+          <Flex flexBasis="25%" flexGrow="1" mr={1}>
+            <Input
+              onBlur={handleBlur}
+              onChange={handleChange}
+              error={touched.state && errors.state}
+              required
+              title="State"
+              name="state"
+            />
+          </Flex>
+          <Flex flexBasis="25%" flexGrow="1">
+            <Input
+              onBlur={handleBlur}
+              onChange={handleChange}
+              error={touched.postal_code && errors.postal_code}
+              required
+              title="Postal code"
+              name="postal_code"
+            />
+          </Flex>
+        </Flex>
+        <Box mt={2}>
+          <Input
+            onBlur={handleBlur}
+            onChange={handleChange}
+            type="tel"
+            error={touched.telephone && errors.telephone}
+            required
+            title="Telephone"
+            name="telephone"
+          />
+        </Box>
+      </Box>
+
+      <Flex mt={4} mb={2} justifyContent="center">
+        <ConditionsOfSaleCheckbox
+          error={touched.agree_to_terms && errors.agree_to_terms}
+          checked={values.agree_to_terms}
+          value={values.agree_to_terms}
+          type="checkbox"
+          name="agree_to_terms"
+          onChange={handleChange}
+          onBlur={handleBlur}
+        />
+      </Flex>
+
+      <Button mt={1} size="large" width="100%" loading={isSubmitting}>
+        Register
+      </Button>
+    </Form>
+  )
+}
+
+const validationSchema = Yup.object().shape({
+  name: Yup.string().required("Name is required"),
+  street1: Yup.string().required("Address is required"),
+  city: Yup.string().required("City is required"),
+  state: Yup.string().required("State is required"),
+  postal_code: Yup.string().required("Postal code is required"),
+  telephone: Yup.string().required("Telephone is required"),
+  agree_to_terms: Yup.bool().oneOf(
+    [true],
+    "You must agree to the Conditions of Sale"
+  ),
+})
+
+export interface RegistrationFormProps
+  extends ReactStripeElements.InjectedStripeProps {
+  relay?: RelayProp
+  sale: AuctionApp_sale
+}
+
+class RegistrationForm extends Component<RegistrationFormProps> {
+  private createBidder(setSubmitting) {
```

Comment:
Note that we could also use the React.FC form:

```tsx
const RegistrationForm: React.FC<RegistrationFormProps> = props => {
  function createBidder(setSubmitting) {
    ...
  }

  function createCreditCard() {
    ...
  }

  return (
    ...
  )
}
```
So as to avoid the use of `this`.